pkg/power/platform/source/redfish.go

```diff
@@ -0,0 +1,286 @@
+	credPath := config.GetRedfishCredFilePath()
+	if credPath == "" {
+		klog.Infof("failed to get redfish credential file path")
+		return nil
+	}
+	if err := nodecred.InitNodeCredImpl(map[string]string{"redfish_cred_file_path": credPath}); err != nil {
+		klog.Infof("%s", fmt.Sprintf("failed to initialize node credential: %v", err))
+		return nil
+	} else {
+		klog.V(5).Infof("Initialized node credential")
+		nodeName := os.Getenv("NODE_NAME")
+		if nodeName == "" {
+			nodeName = "localhost"
+		}
+		redfishCred, err := nodecred.GetNodeCredByNodeName(nodeName, "redfish")
+		if err == nil {
+			userName := redfishCred["redfish_username"]
+			password := redfishCred["redfish_password"]
+			host := redfishCred["redfish_host"]
+			if userName != "" && password != "" && host != "" {
+				klog.V(5).Infof("Initialized redfish credential")
+				probeInterval := config.GetRedfishProbeIntervalInSeconds()
+				interval := time.Duration(probeInterval) * time.Second
+				redfish := &RedFishClient{
+					accessInfo:    RedfishAccessInfo{Username: userName, Password: password, Host: host},
+					systems:       []*RedfishSystemPowerResult{},
+					probeInterval: interval,
+					mutex:         sync.Mutex{},
+				}
+				return redfish
+			}
+		} else {
+			klog.V(1).Infof("%s", fmt.Sprintf("failed to get node credential: %v", err))
+			return nil
+		}
+	}
+	return nil
+}
+
+func (rf *RedFishClient) IsSystemCollectionSupported() bool {
+	system, err := getRedfishSystem(rf.accessInfo)
+
+	if err != nil {
+		klog.Infof("failed to get redfish system info: %v\n", err)
+		return false
+	}
+
+	intervalInMin := 0
+	// iterate each "Members" in the system and get the power info
+	for _, member := range system.Members {
+		// split the OdataID by delimiter "/" and get the system ID
+		split := strings.Split(member.OdataID, "/")
+		if len(split) < 2 {
+			continue
+		}
+		id := split[len(split)-1]
+		res := RedfishSystemPowerResult{}
+		power, err := getRedfishPower(rf.accessInfo, id)
+		if err == nil && len(power.PowerControl) > 0 {
+			res.system = id
+			res.consumedWatts = power.PowerControl[0].PowerConsumedWatts
+			res.timestamp = time.Now()
+			rf.systems = append(rf.systems, &res)
+			klog.V(5).Infof("power info: %+v\n", power)
+			if power.PowerControl[0].PowerMetrics.IntervalInMin > intervalInMin {
+				intervalInMin = power.PowerControl[0].PowerMetrics.IntervalInMin
+			}
+		} else {
+			klog.V(5).Infof("failed to get power info: %v\n", err)
+		}
+	}
+
+	// set a timer to check the power info every probeInterval seconds
+	rf.ticker = time.NewTicker(rf.probeInterval)
+	go func() {
+		for {
+			<-rf.ticker.C
+			for _, system := range rf.systems {
+				power, err := getRedfishPower(rf.accessInfo, system.system)
+				if err == nil && len(power.PowerControl) > 0 {
+					// mutex
+					rf.mutex.Lock()
+					klog.V(5).Infof("power info: %+v\n", power)
+					system.consumedWatts = power.PowerControl[0].PowerConsumedWatts
+					system.timestamp = time.Now()
```

Comment:
Timestamp refreshing is required only when the energy is calculated. It's not here.
```suggestion
```